### ISSUE TITLE
Glitching / Stuttering with Show Note Graph in Workspace

### ISSUE THREAD

**Issue by ScriptAutomate** [dendronhq/dendron#1800]
## Please select the area your bug applies to. (Multiple selections are Possible)

- [ ] Onboard. Initial vault setup and import/export pods
- [ ] Create. Note creation, lookup, snippets and templates
- [x] Retrieve. Backlinks, references, graph view
- [ ] Structure. Refactoring, multi-vault and schemas
- [ ] Publish. Sharing your repo with the world
- [ ] Misc (Choose this if your not sure)

## Describe the bug

Depending on whether the graph is interacted with, or particular steps are taken while the note graph is open, the workspace will glitch/stutter. This causes the open workspace to be unusable unless closing the Note Graph window, and I was unable to run a diagnostics report. I was able to dump the dendron log, though.

> **I'm not sure if I need to open a new issue, but I can't seem to run diagnostics in my workspaces. I get: `error while running command: no workspace file`**

## To Reproduce

1. `Dendron: Show Note Graph'
2. `Ctrl+LeftClick` inside Note Graph window, to highlight and create a multi-selector box
3. Upon selecting multiple notes and letting go, the workspace bugs out

![multi-select-node-graph-bug](https://user-images.githubusercontent.com/5951023/143615137-bf0d230b-bea2-434d-9487-4f48d3617b88.gif)

Another approach:

1. `Dendron: Show Note Graph' (more reproducible in `Show Full Graph` mode, for me)
2. Select a note in the Note Graph
3. Rename the note, placing it within a different hierarchy.
  - Example: `testing.release` -> `testing.ref.release` (where `ref` is a new hierarchy beneath `testing`)
  - This doesn't seem to cause a problem when adding a new sub-hierarchy at the end (ex. `testing.release` -> `testing.release.announce`)

![rename-note-graph-bug](https://user-images.githubusercontent.com/5951023/143615140-3698fa32-cb22-4764-802e-1443de9330b0.gif)

> The glitching/stuttering appears much faster than the gif recorder is able to present

## Expected behavior

No glitching/stuttering is triggered during specific renames or selecting multiple nodes in a note graph.

Could temporary disable the ability to multi-select within the Note Graph? If the multi-select is meant to open multiple notes at once, that should be revisited. If it isn't, the multi-select should be disabled perhaps.

## Desktop (please complete the following information)

- OS: Pop!_OS 21.04
- VSCode/VSCodium: VSCodium v1.62.3
- Dendron Version: v0.70.0

## Additional context

Some reports from users in the community have run into this on Windows, too. Multiple users have experienced, and I was able to reproduce.

## Dendron Log file

Please attach the output of `> Dev:Dendron: Open Logs` here

- [dendron.log](https://github.com/dendronhq/dendron/files/7610091/dendron.log)

**Comment by jonathanyeung** [comment]
I think this is probably the same bug as with the infinite file-open cycling issue we've been discussing with graph view.  Does your repro occur consistently?

**Comment by ScriptAutomate** [comment]
> I think this is probably the same bug as with the infinite file-open cycling issue we've been discussing with graph view. Does your repro occur consistently?

Yeah. I can reproduce in each of my workspaces. Others in the [Dendron Discord have experienced this, also.](https://discord.com/channels/717965437182410783/913532337722101810/913844015810629682). Could definitely be related to the infinite file-open cycling issue, because that's what ends up getting triggered. If this is a duplicate, could combine issues or at least reference each other in them?

**Comment by hikchoi** [comment]
I've disabled the drag-to-select behavior temporarily to prevent this from happening. The current way we pass around messages between VSCode and the webview is causing this problem (multiple active note changes that triggers an infinite chain of messages to rerender the graph / switch active note back and forth), and it should be resolved fundamentally by better messaging, which will come in a bit. Just leaving this here for reference and keeping this issue open for now.
